vm/src/type.h

```diff
@@ -0,0 +1,54 @@
+#ifndef BORGES_TYPE_H
+#define BORGES_TYPE_H
+
+#include <cstddef>
+#include <type_traits>
+#include <vector>
+
+namespace Borges {
+
+class Serializer;
+class Deserializer;
+
+enum class Type : std::size_t {
+    boolean,
+    number,
+    vector_2,
```

Comment:
do we really need vector2/matrix2/matrix3 ? vector3 and matrix4 suffice, albeit with a little bit of redundant data. i suggest we limit the interpreter to using those 2 types only for the geometry calculations.

Reply:
It's not just about redundant data. There are many contexts where it makes sense to restrict yourself to a 2D plane (think offsets, booleans, etc). Having 2D versions of the primitives you need is super-useful, because it prevents you from accidentally doing things like using a 3D curve in a context where a 2D curve is expected.

This may not be very important for us, but it will be very useful for users, since the interpreter can throw a type error if a Curve_3 is used where a Curve_2 is expected. It will also restrict the combinatorial space if we want to mutate existing BOMs.